### PR TITLE
Do not compute Q probability in SiPixel template CPE for iterative tracking

### PR DESCRIPTION
**Pull request by mmasciov**
### PR description:

As per title, using option `speed=0` for SiPixel template CPE with cluster repair (default for phase-1), that disables the computation of Q probability.
In particular, a new instance of transient track recHit builder is defined with option `speed=0`, and only used in main iterative tracking.
At the same time, a number of warnings are commented out, that do not foresee the possibility to not compute Q probability.

### PR validation:

As shown in [report](https://indico.cern.ch/event/1199474/#1-follow-up-on-pixel-cpe-speed), disabling the Q probability computation does not affect physics (zero impact), while it allows to reduce the track final fit time by about 15-20%.

### CODE REVIEW

File: DataFormats/TrackerRecHit2D/interface/SiPixelRecHitQuality.h [hunk 1]
```diff
@@ -65,7 +65,7 @@ class SiPixelRecHitQuality {
     inline float probabilityXY(QualWordType qualWord) const {
       int raw = (qualWord >> probX_shift) & probX_mask;
       if (raw < 0 || raw > 16383) {
-        warningOutOfBoundRaw("XY", raw, qualWord);
+        //warningOutOfBoundRaw("XY", raw, qualWord);
```
Comment (slava77) [inline DataFormats/TrackerRecHit2D/interface/SiPixelRecHitQuality.h:68]: I thought that by default we get 0, but apparently it's not the case. Which values do we get for a not computed case?

Comment (mmasciov) [reply DataFormats/TrackerRecHit2D/interface/SiPixelRecHitQuality.h]: On the one hand, I had commented more warnings than necessary (fixed).
The only relevant warning (left commented) is in the `setProbabilityQ` function.
If not commented, warnings look like:
`ProbQ outside the bounds of the quality word: ProbQ = 1+-2 and QualityWord = 264 --> Now defaulting to ProbQ = 0.0`.
Based on the [definition of the corresponding warning function](https://github.com/trackreco/cmssw/blob/master/DataFormats/TrackerRecHit2D/src/SiPixelRecHitQuality.cc#L66), this means that `prob` equals `-1`, thus the warning is triggered.

**Comment by slava77** [comment]
@mmasciov 
let's make an RFC PR in cms-sw so that the standard tests can be run

File: RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc [hunk 2]
```diff
@@ -504,7 +506,8 @@ void PixelCPEClusterRepair::callTempReco2D(DetParam const& theDetParam,
 
   } else {
     //--- Template Reco succeeded.
-    theClusterParam.hasFilledProb_ = true;
+    if (speed_ < 0)
+      theClusterParam.hasFilledProb_ = true;
```
Comment (slava77) [inline RecoLocalTracker/SiPixelRecHits/src/PixelCPEClusterRepair.cc:510]: apparently 2D reco always calls prob computation, regardless of speed_ value. So, this is a change in behavior

**Pull request merged**